### DISCUSSION TITLE
modules 和 constants 的命名区分

### DISCUSSION THREAD

**Discussion by wewoor** [DTStack/molecule#687] [Ideas]
我们可以把 modules 和 constants 的命名区分下，不然会看起来比较奇怪，这里会怀疑获取的是一个不同的 Constant

_Originally posted by @wewoor in https://github.com/DTStack/molecule/pull/471#discussion_r732653692_

**Comment by mortalYoung** [comment]
另外需要优化的点在于，需要把 modules 做细分，例如有部分的 modules 是以 [xxx,xxx] 作为一个整体的，而我们如果想要禁用某一个 xxx 则会比较困难，需要对其做拆分，会有利于后续禁用内置模块

**Comment by mortalYoung** [comment]
modules 和 constants 命名如下：
- constants 键命名特点：大写、下划线分割、以功能模块起始，如：`EXPLORER_ACTIVITY_ITEM`、`EDITOR_MENU_CLOSE_ALL`
- constants 值命名规则：小写为主，部分以小驼峰法命名、以 `.` 分割、以功能模块起始，如：`editor.closeAll`、`sidebar.explore.saveALL`
- modules 键命名特点：以小写为主，部分以小驼峰法命名，以 `.` 作为分割，如：`menu.file.open`

看一哈 cc @wewoor

**Comment by wewoor** [comment]
没太看懂 modules 部分的命名，如果是下面的这个例子：
```ts
 const { ROOT_FOLDER_CONTEXT_MENU } = this.builtinService.getModules();
```

```ts
 const { folder.context.menu } = this.builtinService.getModules();
```
命名是怎样的

**Comment by mortalYoung** [comment]
考虑如下：
- 在 builtinService 中注册 `ExplorerRegister` `PanelRegister` `MenuRegister` `ActionRegister` `ContextMenuRegister`，目前暂时想到这几个，后续待补充
- 在这些 register 中注册相关的内置模块，而用户可以通过 `molecule.builtin.explorer.inactive()` `molecule.builtin.menu.inactive()` 来禁用相关的内置模块

cc @wewoor
